src/main/java/com/scaler/lld/design/assignments/singleton/FileBasedConfigurationManagerImpl.java

```diff
@@ -1,50 +1,96 @@
 package com.scaler.lld.design.assignments.singleton;
 
 public class FileBasedConfigurationManagerImpl extends FileBasedConfigurationManager {
-
+    private FileBasedConfigurationManagerImpl(){
+        super();
+    }
+    public static FileBasedConfigurationManagerImpl INSTANCE=null;
     @Override
     public String getConfiguration(String key) {
         // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'getConfiguration'");
+        if(INSTANCE!=null){
+            return INSTANCE.properties.getProperty(key);
```

Comment:
You don't need to use the static`INSTANCE` field. `getConfiguration` is an instance method so it will always be called on an object.